lib/MusicBrainz/Server/Controller/Role/Cleanup.pm

```diff
@@ -2,14 +2,30 @@ package MusicBrainz::Server::Controller::Role::Cleanup;
 use Moose::Role;
 use namespace::autoclean;
 
+use MusicBrainz::Server::Constants qw( entities_with );
+
+my %model_to_edit_type = entities_with('add_edit_type', take => sub {
+    my ($type, $properties) = @_;
+    return ($properties->{model} => $properties->{add_edit_type});
+});
+
 after show => sub {
     my ($self, $c) = @_;
     my $entity = $c->stash->{entity};
-    my $eligible_for_cleanup = $c->model( $self->config->{model} )->is_empty($entity->id);
+    my $model = $self->config->{model};
+    my $creation_edit = $c->model('Edit')->find_creation_edit(
+        $model_to_edit_type{$model},
+        $entity->id);
+    # If the creation edit is the only pending edit
+    # and the entity is otherwise empty,
+    # it will be removed as soon as the creation edit passes
+    my $in_cleanup_danger = $entity->edits_pending == 1 &&
+        $creation_edit && $creation_edit->is_open &&
+        $c->model($model)->is_empty($entity->id, 1);
```

Comment:
I would like to make the `1` in `is_empty($entity->id, 1)` documented somehow, but not sure what would be the best way to do this. Just define a `my $ignore_edits = 1;` before and then pass that here?

Reply:
Typically you'd just pass it as a named parameter: add `%args` as the last argument to `is_empty`, pass it as `, ignore_edits => 1`, and access it like `$args{ignore_edits}` inside `is_empty`.